File: Astromaximum/Classes/utils/cells/AmaxSunDegreeCell.m
//
//  AmaxMoonSignCell.m
//  Astromaximum
//
//  Created by admin on 05.12.12.
//  Copyright (c) 2012 S&W Axis. All rights reserved.
//

#import "AmaxSunDegreeCell.h"
#import "Astromaximum-Swift.h"

@implementation AmaxSunDegreeCell

- (void)awakeFromNib {
    [super awakeFromNib];
    degreeLabel = (UILabel *)[self viewWithTag:3];
    zodiacLabel = (UILabel *)[self viewWithTag:4];
}

- (void)configure:(AmaxSummaryItem *)si
{
    [super configure:si];
    AmaxEvent* e = [si mActiveEvent];
    if (e != nil) {
        [timeLabel setText:[e normalizedRangeString]];
        [eventLabel setText:[NSString stringWithFormat:@"%c",
                             getSymbol(TYPE_PLANET, e.mPlanet0)]];
        [degreeLabel setText:[NSString stringWithFormat:@"%d\u00b0",
                             [e getDegree] % 30 + 1]];
        [zodiacLabel setText:[NSString stringWithFormat:@"%c",
                             getSymbol(TYPE_ZODIAC, [e getDegree] / 30)]];
		[self setColorOf:eventLabel byEventMode:e];
    }
    [self updateInfoButtonWith:si];
}

@end

File: Astromaximum/Classes/utils/cells/AmaxSunDegreeCell.swift
//
//  AmaxMoonSignCell.m
//  Astromaximum
//
//  Created by admin on 05.12.12.
//  Copyright (c) 2012 S&W Axis. All rights reserved.
//

import UIKit

class AmaxSunDegreeCell : AmaxTableCell {

    private var degreeLabel:UILabel!
    private var zodiacLabel:UILabel!

    override func awakeFromNib() {
        super.awakeFromNib()
        degreeLabel = (self.viewWithTag(3) as! UILabel)
        zodiacLabel = (self.viewWithTag(4) as! UILabel)
    }

    override func configure(_ extRangeMode: Bool, _ summaryMode: Bool) {
        super.configure(extRangeMode, summaryMode)
        if let events = summaryItem?.mEvents {
            if events.isEmpty {
                return
            }
            let e = events[0]
            if extRangeMode {
                timeLabel?.numberOfLines = 2
                timeLabel?.text =
                    AmaxEvent.long2String(e.date(at: 0), format: AmaxEvent.monthAbbrDayDateFormatter(), h24: false)
                    + "\n"
                    + AmaxEvent.long2String(e.date(at: 1), format: AmaxEvent.monthAbbrDayDateFormatter(), h24: false)
            }
            else {
                timeLabel?.numberOfLines = 1
                timeLabel?.text = e.normalizedRangeString()
            }
            eventLabel?.text = String(format: "%c",
                                      getSymbol(TYPE_PLANET, e.mPlanet0.rawValue))
            degreeLabel.text = String(format: "%d\u{00b0}",
                                 e.getDegree() % 30 + 1)
            zodiacLabel.text = String(format: "%c",
                                      getSymbol(TYPE_ZODIAC, Int32(e.getDegree() / 30)))
            setColorOf(label: timeLabel, byEventMode: e)
        }
    }
}
